src/UI/class-settings-page.php

```diff
@@ -1125,20 +1125,12 @@ private function validate_options_post_type_tracking( array &$input ): void {
 			}
 
 			// Cleanup and sanitized values assignment.
-			unset( $input[ $track_as ] );
 			$input [ $posts ] = self::sanitize_option_array( $temp_posts );
 			$input [ $pages ] = self::sanitize_option_array( $temp_pages );
-		} else {
-			add_settings_error(
-				Parsely::OPTIONS_KEY,
-				$track_as,
-				sprintf(
-					__(
-						'Settings could not be saved because post tracking data is invalid.',
-						'wp-parsely'
-					)
-				)
-			);
+		}
+
+		if ( isset( $input[ $track_as ] ) ) {
```

Comment:
Would it make sense to check if the requires recrawl section is indeed hidden?

Reply:
This needs to be stripped off from the `$input` array in all cases, regardless of the "Requires Recrawl" section's visibility status. This array item is a temporary store for values that we convert to something that all versions of the plugin understand, and we don't want to save it in the database.